### PR TITLE
restore many tag query parameters, begin deprecate ImageURL()

### LINKED ISSUE #900: Posters don't refresh when an image is changed on the server
**Software Versions**
Jellyfin Server Version: Any
Roku Client Version: 1.6.2

**Describe the bug**
PR #605  removed the Tag parameter as it was thought it was causing a flickering issue.  Turns out the Tag parameter is needed in order for caching to work correctly.  To fix we should revert changes from PR #605 

**How To Reproduce**
Update a movie poster on the server.  Notice the Roku client does not update.

**Expected behavior**
Movie poster should update on the client.

**Additional context**
[Reddit report](https://www.reddit.com/r/jellyfin/comments/zmx5dy/regarding_jellyfin_on_roku/)

### PR DESCRIPTION
**Pull request by sevenrats**
manually undo the changes in #605
fixes #900 
these were manual reversions please review carefully.

### CODE REVIEW

**Comment by sevenrats** [comment]
Still needs tags added to image calls:
- [ ] MusicAlbumSongListData.brs
- [ ] MusicArtistData.brs

File: components/data/ChannelData.brs [hunk 1]
```diff
@@ -11,10 +11,10 @@ sub setPoster()
     if m.top.image <> invalid
         m.top.posterURL = m.top.image.url
     else if m.top.json.ImageTags <> invalid and m.top.json.ImageTags.Primary <> invalid
-        imgParams = { "maxHeight": 60 }
-        m.top.hdsmalliconurl = ImageURL(m.top.json.id, "Primary", imgParams)
+        imgParams = { "maxHeight": 60, "Tag": m.top.json.ImageTags.Primary }
+        m.top.hdsmalliconurl = api_API().items.getimageurl(m.top.itemId, "logo", 0, imgParams)
```
Comment (sevenrats) [inline components/data/ChannelData.brs]: wrong

File: components/data/HomeData.brs [hunk 2]
```diff
@@ -16,8 +16,12 @@ sub setData()
     ' Set appropriate Images for Wide and Tall based on type
 
     if datum.type = "CollectionFolder" or datum.type = "UserView"
-        params = { "maxHeight": 261, "maxWidth": 464 }
-        m.top.thumbnailURL = ImageURL(datum.id, "Primary", params)
+        if datum.ImageTags.Primary <> invalid
```
Comment (sevenrats) [inline components/data/HomeData.brs]: Is this check appropriate?

File: components/data/HomeData.brs [hunk 3]
```diff
@@ -28,20 +32,25 @@ sub setData()
         end if
 
     else if datum.type = "Episode"
-        imgParams = { "AddPlayedIndicator": datum.UserData.Played }
+        imgParams = {
+            "AddPlayedIndicator": datum.UserData.Played,
+            "maxHeight": 261,
+            "maxWidth": 464,
+            "Tag": datum.ImageTags.Primary
+        }
```
Comment (sevenrats) [inline components/data/HomeData.brs:36]: replicate this reformatting in similar blocks

File: components/data/HomeData.brs [hunk 4]
```diff
@@ -52,13 +61,19 @@ sub setData()
             imgParams["UnplayedCount"] = datum.UserData.UnplayedItemCount
         end if
 
-        m.top.posterURL = ImageURL(datum.id, "Primary", imgParams)
+        if datum.ImageTags.Primary <> invalid
```
Comment (sevenrats) [inline components/data/HomeData.brs]: do we need this new check either?

File: components/data/HomeData.brs [hunk 5]
```diff
@@ -67,15 +82,21 @@ sub setData()
         imgParams.Append({ "maxHeight": 261 })
         imgParams.Append({ "maxWidth": 175 })
 
-        m.top.posterURL = ImageURL(datum.id, "Primary", imgParams)
+        if datum.ImageTags.Primary <> invalid
```
Comment (sevenrats) [inline components/data/HomeData.brs]: another new check maybe not needed

**Comment by 1hitsong** [comment]
- [ ] Movie Extras throws warning `[sg.scene.bitmap.big] Warning! Loaded texture (800 x 1200) larger than the UI resolution (1920 x 1080) - ExtrasItem/Poster:posterImg`
- [ ] Actor throws warning `[sg.scene.bitmap.big] Warning! Loaded texture (1000 x 1500) larger than the UI resolution (1920 x 1080) - ExtrasItem/Poster:posterImg`
- [ ] TV Show extras throws warning `[sg.scene.bitmap.big] Warning! Loaded texture (1000 x 1500) larger than the UI resolution (1920 x 1080) - ExtrasItem/Poster:posterImg`
- [ ] Clicking on a Music Artist causes crash `ImageURL() is not defined` - line 360 in items.brs
- [ ] Clicking on a song to play it causes crash `PosterImage() is not defined` - line 150 and 154 in itemsbrs

**Review by 1hitsong — CHANGES_REQUESTED** [review]
- [ ] Movie Extras throws warning `[sg.scene.bitmap.big] Warning! Loaded texture (800 x 1200) larger than the UI resolution (1920 x 1080) - ExtrasItem/Poster:posterImg`
- [ ] Actor throws warning `[sg.scene.bitmap.big] Warning! Loaded texture (1000 x 1500) larger than the UI resolution (1920 x 1080) - ExtrasItem/Poster:posterImg`
- [ ] TV Show extras throws warning `[sg.scene.bitmap.big] Warning! Loaded texture (1000 x 1500) larger than the UI resolution (1920 x 1080) - ExtrasItem/Poster:posterImg`
- [ ] Clicking on a Music Artist causes crash `ImageURL() is not defined` - line 360 in items.brs
- [ ] Clicking on a song to play it causes crash `PosterImage() is not defined` - line 150 and 154 in itemsbrs

**Comment by cewert** [comment]
Did you revert #605 or did you change them back manually? I don't see the revert in the commit log

As far as deprecating `ImageURL()`, maybe that should be it's own PR? Or does that need to be done at the same time to make it work?

I approved the CI to run and there were a lot of validation errors - see [here](https://github.com/jellyfin/jellyfin-roku/actions/runs/3722614271/jobs/6414872301). Looks like a lot of the same stuff @1hitsong found. You can also view these locally by running `npm run validate` FYI

**Comment by sevenrats** [comment]
> Did you revert #605 or did you change them back manually? I don't see the revert in the commit log
> 
> As far as deprecating `ImageURL()`, maybe that should be it's own PR? Or does that need to be done at the same time to make it work?
> 
> I approved the CI to run and there were a lot of validation errors - see [here](https://github.com/jellyfin/jellyfin-roku/actions/runs/3722614271/jobs/6414872301). Looks like a lot of the same stuff @1hitsong found. You can also view these locally by running `npm run validate` FYI

i did it manually as I note in my comment.  there were so many conflicts with the automatic revert that it was just easier to reverse it by hand.
as far as the deprecation is concerned, it's something that 1hitsong and I talked about because I'm already knee deep in the relevant lines.  its a burden to review, for sure, but we figured 1 heavy pr would be better than 2 heavy prs, in that respect.

I will validate before I push from now on.

**Comment by sevenrats** [comment]
hmm. actually now that I'm knee-deep in this, I kinda suspect that it should be done differently.  Instead of replacing ImageURL calls with the api call directly, we should simply retool Images.brs to use the api.  the primary reason for this is  the checks in ImageURL:
image

**Comment by sevenrats** [comment]
too much.

**Pull request closed without merging**